test/runtests.jl

```diff
@@ -52,5 +75,48 @@ datadir = joinpath(@__DIR__,"data")
     @test isequal(por, [0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8])
     @test isequal(lit, [1,2,3,4,5,6,7,8])
     @test isequal(sat, [0.8,0.7,0.8,0.7,0.8,0.7,0.8,NaN])
+
+    # test if storing/leading recovers data
+    # to simplify the test, the NaN value is replaced with 99.0
+    sdata[Symbol("Water Saturation")][8] = 99.0
```

Comment:
We don't want to set the value manually, we want to call `load_legacy` and `save_legacy` with the same `na` option. Can you please fix this?

Reply:
It is not that simple. `sdata` is the original point set and still has the NaN value. It may require at least two sequences of save/load to get ride off NaN.

Reply:
I don't understand. Why can't we just `sdata = save_legacy(..., na=whatever)` and then `load_legacy(..., na=whatever)`?

Reply:
because sdata does not change when saving (the na=XXX goes to the file)

Reply:
We don't want sdata to change, right? We want to make sure that saving it to a file and loading it back works regardless of the symbol used for missing values.

Reply:
Perhaps we need to rethink the option `na` to be a pair `na=NaN=>-999` for example, but I don't understand how this simple test can't be written without modifying the `sdata` object.

Reply:
The final solution would be to enhance AbstractData to deal with NaN in the == operation. If you have a suggestion how to do this simple test that would be great.

Reply:
An extra comment: save/load will be easy to test but finishing with `sdata == ndata` will not be possible to test, as sdata remains with the NaN. In that case, we need to test if `ndata` was loaded with the NaN replacement.

Reply:
I see what you are trying to say. The `NaN == NaN` comparison fails so the `values(sdata) == values(ndata)` also fails. I agree we need to improve the missingness situation with `AbstractData` at some point. For now, can we do:

```julia
@test domain(sdata) == domain(ndata)
@test isequal(values(sdata), values(sdata))
```

Does `isequal` solve the comparison with `NaN`? I know it does with scalars `isequal(NaN, NaN) == true` but I didn't test with tables.

Reply:
It seems it does. Very weird semantic of `==` and `isequal`. For any a and b arrays with NaNs:
```
julia> isequal(a, b)
true

julia> a == b
false
```

Let me try with `isequal()`

Reply:
This piece of code works:
```
    na = 99.0
    GslibIO.save_legacy(fname, sdata, na=na)
    ndata = GslibIO.load_legacy(fname, (2,2,2), na=na)
    @test isequal(domain(sdata), domain(ndata))
    @test isequal(values(sdata), values(ndata))
```
Are you happy with it?

Reply:
Yes, much cleaner! Please update the PR accordingly and we can give another round of review tomorrow morning (BRT time). I can take care of the README update after the code is in place.

Reply:
Not a problem